app/javascript/components/common/Editor/TestRuns.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState, useEffect } from 'react'
+import { Submission, SubmissionTestsStatus } from '../Editor'
+import consumer from '../../../utils/action-cable-consumer'
+
+function Content({ submission }: { submission: Submission }) {
+  switch (submission.testsStatus) {
+    case SubmissionTestsStatus.PASS:
+    case SubmissionTestsStatus.FAIL:
+      return (
+        <>
+          {submission.testRuns.map((run) => (
+            <p key={run.name}>
+              name: {run.name}, status: {run.status}, output: {run.output}
+            </p>
+          ))}
+        </>
+      )
+    case SubmissionTestsStatus.ERROR:
+    case SubmissionTestsStatus.OPS_ERROR:
+      return <p>{submission.message}</p>
+    default:
+      return <></>
+  }
+}
+
+export function TestRuns(props: { submission: Submission }) {
+  const [submission, setSubmission] = useState(props.submission)
+
+  useEffect(() => {
+    const subscription = consumer.subscriptions.create(
+      { channel: 'Test::Submissions::TestRunsChannel', id: submission.id },
```

Comment:
We will make the channel name injectable once we move this out into the website.

Reply:
I think this should be the actual channel - I don't think we need any test channels.

I've implemented more of this in https://github.com/exercism/v3-website/pull/193 so if you rebase this onto that, you should have access to `Submission::TestRunsChannel` (note that `Submission` is singular), which can be subscribed to using the `submission.uuid` as you've done here.

I think we're both thinking the same way here 🙂